### PR TITLE
Supporting millisenconds with up to 7 digits.

### PR DESCRIPTION
**Pull request by anderson-joyle**
Issue #895.

### CODE REVIEW

**Comment by LucGenetier** [comment]
✅ No public API change.

**Comment by LucGenetier** [comment]
✅ No public API change.

File: src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryUntypedObject.cs [hunk 1]
```diff
@@ -217,7 +217,11 @@ public static FormulaValue TimeValue_UO(IRContext irContext, UntypedObjectValue[
             if (impl.Type == FormulaType.String)
             {
                 var s = impl.GetString();
-                if (TimeSpan.TryParseExact(s, @"hh\:mm\:ss\.FFF", CultureInfo.InvariantCulture, TimeSpanStyles.None, out TimeSpan res))
+
+                TimeSpan res;
+
+                if (TimeSpan.TryParseExact(s, @"hh\:mm\:ss\.FFFFFFF", CultureInfo.InvariantCulture, TimeSpanStyles.None, out res) ||
```
Comment (MikeStall) [inline src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryUntypedObject.cs:223]: Is this logic shared with DateTimeValue(string)? (And should it be?)

**Comment by LucGenetier** [comment]
✅ No public API change.

File: src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryUntypedObject.cs (hunk 1 above)
Comment (anderson-joyle) [reply src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryUntypedObject.cs:223]: Good question. DateTimeValue and TimeValue functions take locale as arguments. On the other hand, the untyped versions don't and always expect an invariant. That makes me think we don't need to share this logic with DateTimeValue /TimveValue.

**Comment by LucGenetier** [comment]
✅ No public API change.

**Comment by LucGenetier** [comment]
✅ No public API change.

File: src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryUntypedObject.cs [hunk 2]
```diff
@@ -217,7 +217,9 @@ public static FormulaValue TimeValue_UO(IRContext irContext, UntypedObjectValue[
             if (impl.Type == FormulaType.String)
             {
                 var s = impl.GetString();
-                if (TimeSpan.TryParseExact(s, @"hh\:mm\:ss\.FFF", CultureInfo.InvariantCulture, TimeSpanStyles.None, out TimeSpan res))
+
+                if (TimeSpan.TryParseExact(s, @"hh\:mm\:ss\.FFFFFFF", CultureInfo.InvariantCulture, TimeSpanStyles.None, out var res) ||
```
Comment (MikeStall) [inline src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryUntypedObject.cs:221]: Possible future item - can / should this code be shared with the none UO time function?

**Review by MikeStall — APPROVED** [review]

**Comment by LucGenetier** [comment]
✅ No public API change.

**Pull request merged**